src/frollapply.c

```diff
@@ -77,17 +77,21 @@ SEXP memcpyDTadaptive(SEXP dest, SEXP src, SEXP offset, SEXP size) {
 // needed in adaptive=TRUE
 SEXP setgrowable(SEXP x) {
   if (!isNewList(x)) {
-    SET_GROWABLE_BIT(x);
-    SET_TRUELENGTH(x, LENGTH(x)); // important because gc() uses TRUELENGTH to keep counts
+    if (ALTREP(x)) internal_error(__func__, "frollapply's adaptive=T setgrowable should have not been called with an ALTREP input"); // # nocov
+    if (!is_growable(x)) {
+      return make_growable(x);
+    }
+    return x;
   } else {
     // # nocov start ## does not seem to be reported to codecov most likely due to running in a fork, I manually debugged that it is being called when running froll.Rraw
-    for (int i = 0; i < LENGTH(x); i++) {
+    for (R_xlen_t i = 0; i < xlength(x); ++i) {
```

Comment:
We iterate over columns here, not rows. I think int is therefore more appropriate.